ecs/scripts/generate-pr-to-plugins.sh

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env bash
+
+# Constants
+MAPPINGS_SUBPATH="mappings/v8.11.0/generated/elasticsearch/legacy/template.json"
+TEMPLATES_PATH="plugins/setup/src/main/resources/"
+PLUGINS_REPO="wazuh/wazuh-indexer-plugins"
+CURRENT_PATH=$(pwd)
+OUTPUT_PATH=${OUTPUT_PATH:-"$CURRENT_PATH"/../output}
+BASE_BRANCH=${BASE_BRANCH:-master}
+PLUGINS_LOCAL_PATH=${PLUGINS_LOCAL_PATH:-"$CURRENT_PATH"/../wazuh-indexer-plugins}
+# Global variables
+declare -a relevant_modules
+declare -A module_to_file
+
+command_exists() {
+    command -v "$1" &> /dev/null
+}
+
+validate_dependencies() {
+    local required_commands=("docker" "docker-compose" "gh")
+    for cmd in "${required_commands[@]}"; do
+        if ! command_exists "$cmd"; then
+            echo "Error: $cmd is not installed. Please install it and try again."
+            exit 1
+        fi
+    done
+}
+
+fetch_and_extract_modules() {
```

Comment:
`detect_modified_modules()` would be a more straightforward name for this function.

Reply:
Thanks! Applied in commit 67a71bdf9893ffc04da66130e9134440ef3a7229